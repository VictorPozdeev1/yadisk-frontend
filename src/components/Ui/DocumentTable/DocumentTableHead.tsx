import { TableCell, TableHead, TableHeadProps, TableRow, TableRowProps, Typography } from '@mui/material';
import { useTheme } from '@mui/system';
import React, { FC } from 'react';

export interface DocumentTableHeadProps extends TableHeadProps {
  headerNames?: string[];
}

export const DocumentTableHead: FC<DocumentTableHeadProps> = ({ headerNames }) => {
  const theme = useTheme();
  return (
    <TableHead>
      <TableRow>
        {
          headerNames && headerNames?.map(val => (<TableCell colSpan={2} sx={{
            textAlign: 'center',
            borderColor: theme.palette.primary.contrastText,
            borderBottomWidth: '2px',
          }}>
            <Typography variant='h6' component={'h4'}>
              {val}
            </Typography>
          </TableCell>))
        }
      </TableRow>
    </TableHead>
  );
};
