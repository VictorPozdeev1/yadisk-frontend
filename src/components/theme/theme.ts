import { createTheme, responsiveFontSizes } from '@mui/material/styles';
import { Theme } from '@mui/system';

// переопределяем инерфейс брейкпоинтов
declare module '@mui/material/styles' {
  interface BreakpointOverrides {
    xs: false; // removes the `xs` breakpoint
    sm: false;
    md: false;
    lg: false;
    xl: false;
    mobile: true; // adds the `mobile` breakpoint
    tablet: true;
    laptop: true;
    desktop: true;
  }
}
const themeColors={
  fontDark:"#181818",
  actionBlue:'#035FA2',
  secondary:'#EDEDED',

}
export const appTheme:Theme = createTheme({
  typography: {
    fontFamily: 'Manrope, Roboto, Arial',

    h1:{
      fontWeight:'bold',
    },
    h2:{//category title
      fontSize:36,
      lineHeight:1.5,
      color:themeColors.fontDark,
      fontWeight:'bold',
    },
    h3:{
      color:themeColors.fontDark,
      fontWeight:'bold',
    },
    h4:{
      color:themeColors.fontDark,
      fontWeight:'bold',
    },
    h5:{
      color:themeColors.fontDark,
      fontWeight:'bold',
    },
    h6:{//doc table header
      color:themeColors.fontDark,
      fontWeight:'bold',
    },
    body1:{
      color:themeColors.fontDark,
      fontWeight:'600',
    },
  },
  breakpoints:{
    values:{
      mobile:320,
      tablet:805,
      laptop:1200,
      desktop:1523,
    }
  },

  palette: {
    primary: {
      main: themeColors.actionBlue,//action-blue
      light: themeColors.secondary,
      contrastText: themeColors.fontDark,//font-dark
    },
    secondary:{//for table buttons
      main: themeColors.fontDark,//font-dark
    }
  },

});
