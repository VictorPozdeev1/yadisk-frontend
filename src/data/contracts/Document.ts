export {};
interface Document {
    id: string; // Уникальный идентификатор документа
    name: string; // Имя документа
    categoryId: string; // Идентификатор категории, к которой документ принадлежит
    url: string; // URL изображения документа (png)
}
