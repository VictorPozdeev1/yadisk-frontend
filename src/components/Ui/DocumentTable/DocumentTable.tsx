import React, { FC } from 'react'
import Document from '../../../data/contracts/Document';
import { IconButton, Table, TableBody, TableCell, TableHead, TableRow } from '@mui/material';
export interface DocumentTableProps {
  documentList?: Document[];
}
export const DocumentTable: FC<DocumentTableProps> = ({ documentList }) => {
  const tableRows = documentList?.map(row => <DocumentTableRow key={row.resource_id} document={row} />)
  return (
    <Table>
      <TableHead>
        <TableRow>
          <TableCell>Документ</TableCell>
          <TableCell>Категория</TableCell>
        </TableRow>
      </TableHead>
      <TableBody>
        {tableRows ?? tableRows}
      </TableBody>
    </Table>
  )
}


export interface DocumentTableRowProps {
  key: string | number;
  document?: Document;
}
export const DocumentTableRow: FC<DocumentTableRowProps> = ({ key, document }) => {
  return (
    <TableRow>
      <TableCell>
        <IconButton>
          посмотреть документ
        </IconButton>
      </TableCell>
      <TableCell>{'документ' + key}</TableCell>
      <TableCell>{'Категория'}</TableCell>
      <TableCell>
        <IconButton>
          удалить документ
        </IconButton>
      </TableCell>
    </TableRow>
  )
}
