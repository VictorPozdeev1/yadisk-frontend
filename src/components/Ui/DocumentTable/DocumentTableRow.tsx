import React, { FC, useState } from 'react';
import Document from '../../../data/contracts/Document';
import Category from '../../../data/contracts/Category';
import { IconButton, MenuItem, Select, TableCell, TableRow, TableRowProps, Typography } from '@mui/material';
import FileOpenOutlinedIcon from '@mui/icons-material/FileOpenOutlined';
import DeleteForeverRoundedIcon from '@mui/icons-material/DeleteForeverRounded';
import { apiStoreCategories } from '../../../store';
import { toJS } from 'mobx';

export interface DocumentTableRowProps extends TableRowProps {
  document?: Document;
  categoryList?: Category[];//пока не исползуется, берём из mobx
  onDelete?: (arg: unknown) => void;
  onCategoryChange?: (arg: unknown) => void;
  onView?: (arg: unknown) => void;
}
export const DocumentTableRow: FC<DocumentTableRowProps> = ({
  document,
  categoryList,//
  onView,
  onCategoryChange,
  onDelete
}) => {

  const [documentID, setDocumentID] = useState(document?.resource_id);

  const categoryMenu = (categoryList ?? toJS<Category[]>(apiStoreCategories.categories))?.map((item: Category, idx) => (
    <MenuItem
      key={`${idx}_${item.resource_id.slice(-3)}`} value={item.resource_id}
    >
      {item.name}
    </MenuItem>))
  return (
    <TableRow>
      <TableCell>
        <IconButton color='secondary' onClick={(e) => {
          onView && onView({ documentID });
          console.log('view', { documentID });
        }}>
          <FileOpenOutlinedIcon />
        </IconButton>
      </TableCell>

      <TableCell >
        <Typography variant='body1' component='span'>
          {document?.name}
        </Typography>
      </TableCell>

      <TableCell>
        <Select
          variant='standard'
          value={document?.categoryId ?? "-1"}
          onChange={(e) => {
            onCategoryChange && onCategoryChange({
              newCategoryID: e.target.value,
              documentID
            })
            console.log('change category', {
              newCategoryID: e.target.value,
              documentID
            })
          }}
        >
          <MenuItem
            key={'-1'}
            value={'-1'}
          >
            {'не задана категория'}
          </MenuItem>
          {categoryMenu}
        </Select>
      </TableCell>

      <TableCell >
        <IconButton onClick={(e) => {
          onDelete && onDelete({ documentID });
          console.log('delete', { documentID })
        }}>
          <DeleteForeverRoundedIcon color='secondary' />
        </IconButton>
      </TableCell>
    </TableRow>
  );
};
