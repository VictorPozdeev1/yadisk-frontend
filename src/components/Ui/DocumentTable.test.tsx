import { render, screen } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import { DocumentTable } from './DocumentTable/DocumentTable'
import { ThemeProvider } from "@mui/material";
import { appTheme } from '../theme/theme';
import Document from '../../data/contracts/Document';

const list: Document[] = [
    {
        resource_id: 'test_id',
        name: 'test_name',
        categoryId: 'test_category_id',
        file: 'test_file',
        preview: 'test_preview',
        sizes: [{ url: 'test_url' }],
        category: 'test_category',
        path: 'test_path'
    }
]

describe('DocumentTable component', () => {
    test('Render', () => {
        render(
            <BrowserRouter>
                <ThemeProvider theme={appTheme}>
                    <DocumentTable
                        headerNames={['test_category_1', 'test_category_2']}
                        documentList={list}

                    />
                </ThemeProvider>
            </BrowserRouter>
        )
        expect(screen.getByRole('table')).toBeInTheDocument();
        expect(screen.getByText('test_category_1')).toBeInTheDocument();
        expect(screen.getByText('test_category_2')).toBeInTheDocument();
    })

    test('DocumentTable snapshot', () => {
        const documentTable = render(
            <BrowserRouter>
                <ThemeProvider theme={appTheme}>
                    <DocumentTable
                        headerNames={['test_category_1', 'test_category_2']}
                        documentList={list}

                    />
                </ThemeProvider>
            </BrowserRouter>
        )
        expect(documentTable).toMatchSnapshot();
    })

    test('Render without data', () => {
        render(
            <BrowserRouter>
                <ThemeProvider theme={appTheme}>
                    <DocumentTable />
                </ThemeProvider>
            </BrowserRouter>
        )

        expect(screen.getByRole('table')).toBeInTheDocument();
    })
})