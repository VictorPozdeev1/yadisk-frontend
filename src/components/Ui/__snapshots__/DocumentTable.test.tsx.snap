// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentTable component DocumentTable snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiTableContainer-root css-1z12z3u-MuiTableContainer-root"
      >
        <table
          class="MuiTable-root css-hcyzwj-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1x6kfjb-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-ddfke1-MuiTableCell-root"
                colspan="2"
                scope="col"
              >
                <h4
                  class="MuiTypography-root MuiTypography-h6 css-112w9ep-MuiTypography-root"
                >
                  test_category_1
                </h4>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-ddfke1-MuiTableCell-root"
                colspan="2"
                scope="col"
              >
                <h4
                  class="MuiTypography-root MuiTypography-h6 css-112w9ep-MuiTypography-root"
                >
                  test_category_2
                </h4>
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-1xq2fmy-MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root css-s6jnxt-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-n8l7vr-MuiTableCell-root"
              >
                <a
                  href="/test_category/test_id"
                  title="смотреть документ"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary MuiIconButton-sizeMedium css-xbcopp-MuiButtonBase-root-MuiIconButton-root"
                    data-testid="view_btn"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="FileOpenOutlinedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15 22H6c-1.1 0-2-.9-2-2V4c0-1.1.9-2 2-2h8l6 6v6h-2V9h-5V4H6v16h9v2zm4-.34v-2.24l2.95 2.95 1.41-1.41L20.41 18h2.24v-2H17v5.66h2z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </a>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-17pjao6-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-k6gp7g-MuiTypography-root"
                  title="test_name"
                >
                  test_name
                </p>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-13zozh8-MuiTableCell-root"
              >
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary  css-f8s8ul-MuiInputBase-root-MuiInput-root-MuiSelect-root"
                  data-testid="test_select"
                  title="сменить категорию"
                >
                  <div
                    aria-controls=":r1:"
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
                    role="combobox"
                    tabindex="0"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </div>
                  <input
                    aria-hidden="true"
                    aria-invalid="false"
                    class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                    tabindex="-1"
                    value="test_category"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </div>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-n8l7vr-MuiTableCell-root"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary MuiIconButton-sizeMedium css-xbcopp-MuiButtonBase-root-MuiIconButton-root"
                  data-testid="delete_btn"
                  tabindex="0"
                  title="удалить документ"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="DeleteForeverRoundedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm3.17-6.41a.9959.9959 0 0 1 0-1.41c.39-.39 1.02-.39 1.41 0L12 12.59l1.41-1.41c.39-.39 1.02-.39 1.41 0s.39 1.02 0 1.41L13.41 14l1.41 1.41c.39.39.39 1.02 0 1.41s-1.02.39-1.41 0L12 15.41l-1.41 1.41c-.39.39-1.02.39-1.41 0a.9959.9959 0 0 1 0-1.41L10.59 14l-1.42-1.41zM18 4h-2.5l-.71-.71c-.18-.18-.44-.29-.7-.29H9.91c-.26 0-.52.11-.7.29L8.5 4H6c-.55 0-1 .45-1 1s.45 1 1 1h12c.55 0 1-.45 1-1s-.45-1-1-1z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiTableContainer-root css-1z12z3u-MuiTableContainer-root"
    >
      <table
        class="MuiTable-root css-hcyzwj-MuiTable-root"
      >
        <thead
          class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-1x6kfjb-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-ddfke1-MuiTableCell-root"
              colspan="2"
              scope="col"
            >
              <h4
                class="MuiTypography-root MuiTypography-h6 css-112w9ep-MuiTypography-root"
              >
                test_category_1
              </h4>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-ddfke1-MuiTableCell-root"
              colspan="2"
              scope="col"
            >
              <h4
                class="MuiTypography-root MuiTypography-h6 css-112w9ep-MuiTypography-root"
              >
                test_category_2
              </h4>
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root css-1xq2fmy-MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root css-s6jnxt-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-n8l7vr-MuiTableCell-root"
            >
              <a
                href="/test_category/test_id"
                title="смотреть документ"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary MuiIconButton-sizeMedium css-xbcopp-MuiButtonBase-root-MuiIconButton-root"
                  data-testid="view_btn"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="FileOpenOutlinedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15 22H6c-1.1 0-2-.9-2-2V4c0-1.1.9-2 2-2h8l6 6v6h-2V9h-5V4H6v16h9v2zm4-.34v-2.24l2.95 2.95 1.41-1.41L20.41 18h2.24v-2H17v5.66h2z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </a>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-17pjao6-MuiTableCell-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-k6gp7g-MuiTypography-root"
                title="test_name"
              >
                test_name
              </p>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-13zozh8-MuiTableCell-root"
            >
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary  css-f8s8ul-MuiInputBase-root-MuiInput-root-MuiSelect-root"
                data-testid="test_select"
                title="сменить категорию"
              >
                <div
                  aria-controls=":r1:"
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
                  role="combobox"
                  tabindex="0"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </div>
                <input
                  aria-hidden="true"
                  aria-invalid="false"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  tabindex="-1"
                  value="test_category"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-n8l7vr-MuiTableCell-root"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary MuiIconButton-sizeMedium css-xbcopp-MuiButtonBase-root-MuiIconButton-root"
                data-testid="delete_btn"
                tabindex="0"
                title="удалить документ"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="DeleteForeverRoundedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm3.17-6.41a.9959.9959 0 0 1 0-1.41c.39-.39 1.02-.39 1.41 0L12 12.59l1.41-1.41c.39-.39 1.02-.39 1.41 0s.39 1.02 0 1.41L13.41 14l1.41 1.41c.39.39.39 1.02 0 1.41s-1.02.39-1.41 0L12 15.41l-1.41 1.41c-.39.39-1.02.39-1.41 0a.9959.9959 0 0 1 0-1.41L10.59 14l-1.42-1.41zM18 4h-2.5l-.71-.71c-.18-.18-.44-.29-.7-.29H9.91c-.26 0-.52.11-.7.29L8.5 4H6c-.55 0-1 .45-1 1s.45 1 1 1h12c.55 0 1-.45 1-1s-.45-1-1-1z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
